app1/next.config.js

```diff
@@ -34,7 +13,18 @@ module.exports = {
           "./app1": "./pages/index.tsx",
         },
         remotes: getRemotes(isServer),
-        shared: {},
+        shared: {
+          react: {
+            requiredVersion: pkg.dependencies.react,
+            singleton: true,
+            strictVersion: true,
+          },
+          "react-dom": {
+            requiredVersion: pkg.dependencies["react-dom"],
+            singleton: true,
+            strictVersion: true,
+          },
+        },
```

Comment:
Could we use this strategy for internal global dependencies, or is it better to have them as a micro-frontend?

Reply:
I think it's better to have this in an MFE for now, so that anytime a team would like to host another MFE, they wouldn't have to worry about setting this up.

What we can do to avoid this boilerplate for teams is make this part of the startup repo template that we will provide for migration.